jvm/src/main/java/com/microsoft/bond/protocol/BinaryStreamReader.java

```diff
@@ -0,0 +1,81 @@
+package com.microsoft.bond.protocol;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Responsible for reading Bond primitive data types from an input stream.
+ * This class encapsulates little-endian to big-endian conversion.
```

Comment:
It's also responsible for String encodings. I see you've got a helper for that, but from `BinaryStreamReader`'s callers' perspectives, `BinaryStreamReader` is responsible for this.